### ISSUE TITLE
Support native True/False properties?

### ISSUE THREAD

**Issue by fjdiazt** [Microsoft/vsts-extension-toggle-control#13]
I know the control only supports boolean fields, but I'm wondering if it is at all possible to support all VSTS true/false fields in the future, otherwise its use is extremely limited.

**Comment by mmanela** [comment]
It's possible but we would need to define what the expected behavior is. Do we assume the field has true and false values? What if other values appear?

**Comment by fjdiazt** [comment]
I don't have any experience with the API but I would assume each field as a default value, if there is no default value, is there a way to check for that?  Otherwise I would allow the user to specify the default value when selecting the field.

As for other values, I think it's best to allow fields that have only 2 possible values, true/false, yes/no. However, if there is no easy way to be sure of possible field values, then lock the control (gray it out) and notify the user a 3rd non-expected value was supplied. The user would then have to either remove the control from the form or change the mapped field.

**Comment by mmanela** [comment]
@ostreifel and @mohitbagra  Thoughts on this idea?

**Comment by cschleiden** [comment]
The control used to support this, this change https://github.com/Microsoft/vsts-extension-toggle-control/commit/ba77c860e509f1e3e7b032d28bde220828e453bc took it away.. not sure why but it shouldn't be hard to bring back.

**Comment by fjdiazt** [comment]
Well the change was quite deliberate, there must be a reason behind it, any idea why it was removed?

**Comment by mmanela** [comment]
Yes, I believe I remember now. The admin UI screen to configure the control was confusing when it supported both Boolean and String. We had user feedback on this and decided to simplify it to only boolean.

**Comment by cschleiden** [comment]
That killed a very valid scenario, in my opinion. Boolean support in web access isn't that old, and many people are using yes/no picklists, that now cannot be easily replaced with the toggle control.

**Comment by mmanela** [comment]
I agree it is a valid scenario. But not sure if it makes sense to confuse one scenario for another.

**Comment by cschleiden** [comment]
I would argue the scenario I described is more important to be covered by an extension than the one that's currently supported. Displaying a boolean field as a toggle instead of a checkbox should be done by a native control. More and more experiences in VSTS are adopting the toggle control and moving away from checkboxes, why should the WIT form be any different?

**Comment by mmanela** [comment]
That is a different conversation. I agree the native control should be a toggle. Lets take that up with the feature team PMs. @labrose @joserady

**Comment by cschleiden** [comment]
No, it's related. I don't think we can cover rendering a "yes/no" picklist as a toggle from the product, but that's a great use case for an extension. That's what it was originally designed for.
